lib/syskit/data_service.rb

```diff
@@ -90,10 +90,13 @@ def find_device_attached_to(service = nil)
                     elsif driver_services.size > 1
                         raise ArgumentError, "#{self} handles more than one device, you must specify one of #{driver_services.map(&:name).sort.join(", ")} explicitely"
                     end
-                    service = driver_services.first
+                    service = driver_services.find do |driver_service|
+                        service == driver_service
+                    end
+                    
                 end
 
-                arguments["#{service.name}_dev"]
```

Comment:
To clarify things: the change of allowing starting a driver if at least one device is assigned is fine by me.

I have a problem with the renaming of the argument to 'dev'. You could before have one TaskContext be *at the same time* driver for two (or more) devices. It has been used (I added it explicitely because the feature was needed), so I would feel bad about removing it.